kratos/includes/define.h

```diff
@@ -693,46 +693,58 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
 //
 //-----------------------------------------------------------------
 
+#ifdef KRATOS_SET_REGISTER_SOURCE
+#undef KRATOS_SET_REGISTER_SOURCE
+#endif
+#define KRATOS_SET_REGISTER_SOURCE(name) \
+    std::string mApplicationName = name;
+
 #ifdef KRATOS_REGISTER_GEOMETRY
 #undef KRATOS_REGISTER_GEOMETRY
 #endif
 #define KRATOS_REGISTER_GEOMETRY(name, reference) \
     KratosComponents<Geometry<Node>>::Add(name, reference); \
+    KratosComponents<Geometry<Node>>::AddSource(mApplicationName, name); \
```

Comment:
thats rather dirty :sweat_smile: 
especially for the core

I cannot think of an easy solution without changing the interface, but please comment this properly. This can lead to nasty errors if used outside the application context

How about moving these macros to `kratos/includes/kratos_application.h`, then this would be much cleaner, and should not break existing code.
What do you think?

Reply:
On second thought, I would even go further and not only move this, but also replace the macros with member functions of `KratosApplication`. This way it cannot be used wrong

This can be done in a separate PR ofc, but just mentioning what I would imagine the final solution for this to be

Reply:
I meant to move only the macros for the types that always belong to an application, like elements and conditions.

Those are implicitly forced to belong to an application anyway.

The other generic macros cannot be moved ofc

Regarding the registration of the core things, do you think it would make sense to move those to the Register fct of the core?

Reply:
The idea is to move everything to the registry and remove KratosComponents, eventually...

Reply:
If you preffer, instead of using `mApplicationName` as the key I can use the mangled name of the class executing the macro (as a matter of fact, that was my first approach). That way no compilation errors should ever arise, but it makes deregistering things much more cumbersome as you have to ensure that the register and deregister function exists in an isolated context and its consistent across the application :S

Reply:
Hm I see

Well if you have a global plan then no complaints from my side 😄